hedera-mirror-rest/viewmodel/contractResultDetailsViewModel.js

```diff
@@ -42,21 +44,69 @@ class ContractResultDetailsViewModel extends ContractResultViewModel {
    * @param {Transaction} transaction
    * @param {ContractLog[]} contractLogs
    * @param {ContractStateChange[]} contractStateChanges
+   * @param {FileData} fileData
    */
-  constructor(contractResult, recordFile, transaction, contractLogs, contractStateChanges) {
+  constructor(contractResult, recordFile, transaction, contractLogs, contractStateChanges, fileData) {
     super(contractResult);
-    this.block_hash = utils.addHexPrefix(recordFile.hash, false);
+
+    this.block_hash = utils.addHexPrefix(recordFile.hash);
     this.block_number = recordFile.index;
-    this.hash = utils.toHexString(transaction.transactionHash, true);
     this.logs = contractLogs.map((contractLog) => new ContractLogResultsViewModel(contractLog));
     this.result = TransactionResult.getName(transaction.result);
+    this.transaction_index = transaction.index;
+    this.nonce = transaction.nonce;
+
     this.state_changes = contractStateChanges.map(
       (contractStateChange) => new ContractResultStateChangeViewModel(contractStateChange)
     );
     this.status =
       transaction.result === ContractResultDetailsViewModel._FAIL_PROTO_ID
         ? ContractResultDetailsViewModel._SUCCESS_RESULT
         : ContractResultDetailsViewModel._FAIL_RESULT;
+
+    // default eth related values
+    this.access_list = null;
+    this.block_gas_used = null;
+    this.chain_id = null;
+    this.gas_price = null;
+    this.max_fee_per_gas = null;
+    this.max_priority_fee_per_gas = null;
+    this.r = null;
+    this.s = null;
+    this.type = null;
+    this.v = null;
+
+    if (`${transaction.type}` === TransactionType.getProtoId('ETHEREUMTRANSACTION')) {
+      this.access_list = transaction.accessList ? utils.addHexPrefix(transaction.accessList) : null;
+      this.amount = !_.isNil(transaction.value)
+        ? parseInt(Buffer.from(transaction.value, 'utf8').toString(), 16)
+        : null;
+      this.block_gas_used = recordFile.gasUsed;
+      this.chain_id = transaction.chainId ? utils.addHexPrefix(transaction.chainId) : null;
+      this.from = !_.isNil(contractResult.senderId) ? EntityId.parse(contractResult.senderId).toEvmAddress() : null;
+      this.gas_limit = !_.isNil(transaction.gasLimit) ? transaction.gasLimit : null;
```

Comment:
This should coalesce with the gas_limit set in the super.
```suggestion
      if (!_.isNil(transaction.gasLimit) {
        this.gas_limit = transaction.gasLimit;
      }
```

Reply:
Done.